### PR TITLE
CombinedManagers 1.0

### PR DESCRIPTION
**Pull request by VinceMack**
**Game Managers Scene**
The goal of this scene is to combine active development paths, providing a solid starting point beyond DP2. Efforts have been made to ensure clean code with comments and adherence to standard C# practices.

The Game Managers scene incorporates:

- Tyler's work on pathfinding, grid, and tilemap
- Vincent's contributions to labor order manager, pawn, and labor orders
- Connor's OOP structuring and refinement of labor orders

A primary GameManager GameObject (GO) is positioned at the root of all other GOs. Within GameManager, there are Pawns GO and Grid GO, which serve as containers for pawns and the grid/tilemap.

**NOTE**: To avoid interfering with ongoing development, "_VM" versions of some classes (e.g., Pawn_VM.cs) have been created. These can be replaced with regular versions upon approval.

The **GameManager** GO contains four scripts:

1. **LaborOrderManager**: Generates a random set of labor orders and assigns them to available pawns.
2. **InputManager**: Handles user input. Currently, users can click the screen to obtain tile information.
3. **GridManager**: Sets up the grid and associated tilemap (NOTE: current map generation relies on simple RNG and serves as a testing example).
4. **PathfindingManager**: Creates paths for pawns to follow when completing assigned work orders.

Additional scripts used in this scene:

**Pawn**: Receives labor orders from LaborOrderManager, uses PathfindingManager to locate its position, moves to that location, executes the assigned labor order, and then returns to LaborOrderManager for further assignment. This script is applied to the Pawn prefab.

Changes have been made to the LaborOrderManager in this PR. Interactions with other scripts have been commented out until approval of all PR changes. Subsequently, we can rename the "_VM" versions and resolve interactions with the LaborOrderManager.

### CODE REVIEW

**Comment by VinceMack** [comment]
Tyler, the primary thing I would like you to review is the GridManager and PathfindingManager classes because these classes are directly derivative of your work (mostly unchanged). I just made small adjustments because I had my Tilemap at a 0.5x0.5x0 anchor to align the tiles with the grid, and I was using GetCellCenterWorld() to center Pawns at the offset. My Tilemap starts generating a 0,0 in the bottom left hand side and then populated to positive values only. I think you did it a bit differently, so I hade to change how the functions iterate over the bounds of the map, but he pathfinding itself is completely unchanged.

**Comment by VinceMack** [comment]
Connor, I would like you to review the changes with the LaborOrder_VM (and the derived classes) that I modeled after the OOP changes you made the the first version of the LaborOrderManager. Similarly, I would like you to overview the Pawn_VM class to assure that the way the pawn executes the labor order makes sense to you and seems good.

**Comment by VinceMack** [comment]
Like I said, I want to make this a really clean and comprehensive combination of most of our developments efforts so far so we have a clean starting point after DP2.

**Comment by VinceMack** [comment]
Auto testing is still broken, but I have manually confirmed this PR does not fail any current tests.

File: Assets/Scripts/GameManagers/LaborOrders/LaborOrder_Woodcut_VM.cs
Comment (Syrodai) [inline Assets/Scripts/GameManagers/LaborOrders/LaborOrder_Woodcut_VM.cs:1]: The original file will have to be changed to work but there should be more here.

Comment (Syrodai) [reply Assets/Scripts/GameManagers/LaborOrders/LaborOrder_Woodcut_VM.cs:1]: I can adapt it. Don't worry about fixing this.

File: Assets/Scripts/GameManagers/LaborOrderManager.cs [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    }
+
+    // Update method to assign labor tasks to available pawns
+    void Update()
+    {
+        if (availablePawns.Count > 0 && getNumOfLaborOrders() > 0)
+        {
+            assignPawn();
+        }
+    }
```
[the remark this answers is not in the document]
Comment (VinceMack) [reply Assets/Scripts/GameManagers/LaborOrderManager.cs:147]: That's a good point, and I fail to see how it wouldn't be more performant to just assign sequentially in one frame all the possible assignments. I will make this change.

File: Assets/Scripts/GameManagers/Pawn_VM.cs [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+        pawnName = "Pawn" + ++pawnCount;
+        name = pawnName;
+
+        isAssigned = false;
+    }
+
+    // Update function for handling logic
+    void Update()
+    {
+        if (isAssigned)
```
[the remark this answers is not in the document]
Comment (VinceMack) [reply Assets/Scripts/GameManagers/Pawn_VM.cs:214]: You are correct. I might end up adding a new method in the Pawn class assignLaborOrder() which can be utilized by the LaborOrderManager or other systems to directly set the labor order for some Pawn while keeping all the necessary variables set for the logic (i.e. isAssigned).

I will make this change or something similar.

File: Assets/Scripts/GameManagers/LaborOrderManager.cs [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+    // Start method to generate and queue random labor tasks
+    void Start()
+    {
+        // generate 100 labor orders with random labor types and random ttc's and add them to the appropriate queue in the array of labor order queues
+        for (int i = 0; i < 100; i++)
+        {
+            addLaborOrder(new LaborOrder_Base_VM(true));
+        }
```
[the remark this answers is not in the document]
Comment (VinceMack) [reply Assets/Scripts/GameManagers/LaborOrderManager.cs:137]: Yeah, that was an oversight on my part. Will revise.

**Comment by cookt5** [comment]
Looks good, nice job!

File: Assets/Scripts/GameManagers/Pawn_VM.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (VinceMack) [reply Assets/Scripts/GameManagers/Pawn_VM.cs:214]: That's a great point, and the fix should be simple. Will revise.

File: Assets/Scripts/GameManagers/LaborOrderManager.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (VinceMack) [reply Assets/Scripts/GameManagers/LaborOrderManager.cs:163]: **REVISED**: The logic inside of `assignPawn()` has been updated to exhaust all possible assignments by using a while loop controlled by the same logic controlling the call to `assignPawn()` in `Update()`. The assignment logic has been slightly adjusted to accommodate this change.

File: Assets/Scripts/GameManagers/Pawn_VM.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (VinceMack) [reply Assets/Scripts/GameManagers/Pawn_VM.cs:215]: **REVISED**: The `Pawn` class has been updated to disallow the assignment of a new order when `isAssigned` is `true`. Additionally, `isAssigned` will no longer be set to `false` until after the coroutine in `Update()` has finished and the call stack returns to  `Update()`.

File: Assets/Scripts/GameManagers/LaborOrderManager.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (VinceMack) [reply Assets/Scripts/GameManagers/LaborOrderManager.cs:150]: **REVISED**: The duplicate addition of labor orders has been removed from the constructor.

**Pull request merged**